### PR TITLE
⬆️ 0.8/pydantic2 fix

### PR DESCRIPTION
**Pull request by s-weigand**
This is based on #1368 but I made a separate PR so @joernweissenborn can review my changes.

Upgrading to pydantic 2 is the first step we need to take before we can make schema generation work.

### CODE REVIEW

File: glotaran/model/item.py [hunk 1]
```diff
@@ -38,6 +38,34 @@
 META_VALIDATOR = "__glotaran_validator__"
 
 
+class GlotaranFieldMetadata(UserDict):
```
Comment (s-weigand) [inline glotaran/model/item.py:41]: I know this whole thing isn't really pretty but we need to it keeps us going for now until we reevaluate the need for an extra evaluation mechanism.

**Comment by s-weigand** [comment]
We still need to deal with
```
glotaran/plugin_system/base_registry.py:130
glotaran/plugin_system/base_registry.py:130
glotaran/plugin_system/base_registry.py:130
glotaran/plugin_system/base_registry.py:130
glotaran/plugin_system/base_registry.py:130
glotaran/plugin_system/base_registry.py:130
  /home/runner/work/pyglotaran/pyglotaran/glotaran/plugin_system/base_registry.py:130: DeprecationWarning: SelectableGroups dict interface is deprecated. Use select.
    for entry_points in [entry_points[p] for p in plugin_names]:  # type:ignore[assignment]

../../../../../opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/pydantic/_internal/_fields.py:128
  /opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/pydantic/_internal/_fields.py:128: UserWarning: Field "model_interval" has conflict with protected namespace "model_".
  
  You may be able to resolve this warning by setting `model_config['protected_namespaces'] = ()`.
    warnings.warn(

../../../../../opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/pydantic/_internal/_fields.py:163
../../../../../opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/pydantic/_internal/_fields.py:163
  /opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/pydantic/_internal/_fields.py:163: UserWarning: Field name "register_as" shadows an attribute in parent "Element"; 
    warnings.warn(

../../../../../opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/pydantic/_internal/_fields.py:163
../../../../../opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/pydantic/_internal/_fields.py:163
../../../../../opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/pydantic/_internal/_fields.py:163
  /opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/pydantic/_internal/_fields.py:163: UserWarning: Field name "data_model_type" shadows an attribute in parent "Element"; 
    warnings.warn(

../../../../../opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/pydantic/_internal/_fields.py:163
  /opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/pydantic/_internal/_fields.py:163: UserWarning: Field name "data_model_type" shadows an attribute in parent "ExtendableElement"; 
    warnings.warn(

../../../../../opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/pydantic/_internal/_fields.py:163
  /opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/pydantic/_internal/_fields.py:163: UserWarning: Field name "register_as" shadows an attribute in parent "ExtendableElement"; 
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
```
But those are for a different discussion and PR 😅

**Review by jsnel — APPROVED** [review]
Changes look sound to me

**Review by joernweissenborn — APPROVED** [review]
Lgtm

**Pull request merged**